var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -106,7 +106,14 @@ class Mesa(MesonPackage):
 
     # Provides
     provides("libglx", when="+glx")
-    # provides('egl@1.5', when='+egl')
+    provides("egl@1.5", when="+egl")
```

Comment:
The virtual should be renamed `libegl` as there is already an `egl` package that this will conflict with.

The EGL package should also be updated here to provide `libegl` and `gl` to reflect the current hack there that detects both for system EGL. It is also possible that we could just drop the hack completely with the `libegl` virtual.

Then we also need to update the references in the packages that are currently EGL aware. 
* GLEW
* ParaView